### PR TITLE
@ #404 | should add dev-setup for teracy-dev

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #404

### CODE REVIEW

**Comment by hieptranquoc** [comment]
@hoat le please help me check this pr!

File: .gitignore [hunk 1]
```diff
@@ -136,8 +136,10 @@ Session.vim
 !.travis
 
 # docs
-_build
```
Comment (hoatle) [inline .gitignore]: _build/*
!.gitkeep

File: .travis/build.sh [hunk 2]
```diff
@@ -9,11 +9,16 @@ if [[ $BUILD_TYPE == "dev" ]]; then
     docker run --rm -v $(pwd):/opt/app $CONTAINER_IMAGE rake build
     docker run --rm -v $(pwd):/opt/app $CONTAINER_IMAGE bundle exec rspec
 elif [[ $BUILD_TYPE == "docs" ]]; then
-    cd docs
-    docker_build
-    docker run -v $(pwd):/opt/app $CONTAINER_IMAGE make generate
-    cd ..
+    if [[ $TRAVIS_BRANCH == 'master' ]] ; then
```
Comment (hoatle) [inline .travis/build.sh]: no, don't touch this

File: dev-setup/README.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
+# Teracy-dev-setup
+
+Please follow this getting started guide to set up the development environment.
+
+Please take a cup of coffee with you, you mostly don't have to do anything but wait for the result, enjoy!
+
+
+## Getting Started
+
+- Set up `teracy-dev` if not yet, follow: http://dev.teracy.org/docs/develop/getting_started.html .
```
Comment (hoatle) [inline dev-setup/README.md]: duplicated of https://github.com/teracyhq/dev-setup/blob/develop/README.md

File: dev-setup/README.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+- You need to fetch the latest changes of `teracy-dev` before going to the next step.
```
Comment (hoatle) [inline dev-setup/README.md]: not a very logic user workflow

File: docs/Dockerfile [hunk 5]
```diff
@@ -1,16 +1,43 @@
-FROM nginx:1.11
+ARG  PYTHON_VERSION=2.7
```
Comment (hoatle) [inline docs/Dockerfile]: let's use PYTHON_VERSION=3.6

**Review by hoatle — DISMISSED** [review]

**Comment by hoatle** [comment]
@phuonglm please review

File: .gitignore (hunk 1 above)
Comment (hoatle) [reply .gitignore]: sorry, this works:

```
docs/_build/**
!.gitkeep
```

Comment (hieptranquoc) [reply .gitignore]: fixed!

File: .travis/build.sh (hunk 2 above)
Comment (hieptranquoc) [reply .travis/build.sh]: thanks you , fixed!

File: dev-setup/README.md (hunk 3 above)
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md (hunk 4 above)
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: docs/Dockerfile (hunk 5 above)
Comment (hieptranquoc) [reply docs/Dockerfile]: updated!

**Comment by hieptranquoc** [comment]
@hoatle please help me check this PR!

**Comment by hoatle** [comment]
assigned to @phuonglm

**Comment by phuonglm** [comment]
This PR is ok for me, @hoavt please review https://github.com/teracyhq/dev/pull/407/files#diff-205ebbec5c138cdbef078a3f0edcd650 for me.

**Review by hoavt — COMMENTED** [review]
Please check the comments @hieptranquoc

File: dev-setup/README.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
+# Teracy-dev-setup
+
+Please follow this getting started guide to set up the development environment.
+
+Please take a cup of coffee with you, you mostly don't have to do anything but wait for the result, enjoy!
+
+
+## Getting Started
+
+- Please follow document at https://github.com/teracyhq/dev-setup/blob/develop/README.md to set up teracy-dev and teracyhq/dev-setup.
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+
+## Setting up teracy-dev-setup
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+- Clone the forked repo into the `~/teracy-dev/workspace` directory.
+
+    ```bash
+    $ cd ~/teracy-dev/workspace/
+    $ git clone <your_forked_repository_here> dev
+    $ cd dev
+    $ git remote add upstream git@github.com:teracyhq/dev.git
+    ```
+
+- Add configuration path to `vagrant_config_default.json` in `teracyhq/dev-setup`, if is not exist. It should look like similar this:
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+- http://dev-docs.<vm_ip>.xip.io to check out the docs within your LAN network.
+
+- http://ngrok-prod.dev-docs.teracy.dev to check out the docs on the Internet.
+
+
+### Local review mode docs
+
+To review work and PRs submitted by others, for example, with
+`hoatle/teracy-dev-docs:improvements-176-teracy-dev-docs-guide` Docker image, run it:
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+
+## travis-ci configuration
+
+You just need to configure travis-ci only one time. After each travis-ci build, new Docker images
+are pushed, we can review your work (PR) by running the Docker images instead of fetching git code
+and build it on local ourselves.
+
+Here are things you need to do:
+
+- Register your account at https://hub.docker.com
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+- Register your account at travis-ci.org
+- Enable teracy-dev repository on travis-ci (for example: https://travis-ci.org/hoatle/teracy-dev)
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+- Fill in the following environment variables settings for teracy-dev travis-ci project by
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/README.md [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+  following: https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings.
+  In the *Name* and *Value* fields, please add the info below correlatively:
+
+  + Fill in "DOCKER_USERNAME" into the *Name* field, and your Docker username into the *Value*  field
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/README.md]: fixed!

File: dev-setup/chef/README.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
+# teracy-dev-setup cookbook
+
+Chef cookbooks to manage teracy-dev VM for the project.
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/chef/README.md]: fixed!

File: dev-setup/chef/README.md [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+You can use other software provisioning tools other than Chef, see: https://www.vagrantup.com/docs/provisioning/
+`teracy-dev` supports all the configuration of vagrant supported provisioners into the
+`vagrant_config_override.json` file, you can combine them for the dev-setup.
+
+
+## dev-setup/chef/main-cookbooks/teracy-dev-setup cookbook
+
+- This Chef project is used to manage the VM for the `teracy-dev` project.
+
+- The `teracy-dev-setup` Chef project was initial generated by the following commands:
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/chef/README.md]: fixed!

File: dev-setup/chef/README.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+$ vagrant ssh
+$ ws
+$ cd dev/dev-setup/chef
+$ docker-compose up
+```
+
+and then `$ vagrant rsync-back` from the host terminal to sync generated files back to the host.
+
+
+## Configure the teracy cookbook to be used with `teracy-dev`
```
[the remark this answers is not in the document]
Comment (hieptranquoc) [reply dev-setup/chef/README.md]: fixed!

**Comment by hieptranquoc** [comment]
@hoavt i updated! please help me check this PR

**Review by hoavt — DISMISSED** [review]
It's ok to merge

File: docs/docker-compose.yml [hunk 17]
```diff
@@ -1,13 +1,32 @@
...
+    image: ${APP_DEV_IMAGE:-python:3.6}
+    working_dir: /opt/app
+    command: sh run-dev.sh
     volumes:
       - .:/opt/app
+    environment:
+      VIRTUAL_HOST: ${APP_DEV_VIRTUAL_HOST:-dev.dev-docs.teracy.dev, ~^dev.dev-docs\..*\.xip\.io}
+      HTTPS_METHOD: noredirect # support both http and https
+    ports:
+      - "8000:8000"
```
Comment (hoatle) [inline docs/docker-compose.yml]: no fixed port mapping here

**Review by hoatle — CHANGES_REQUESTED** [review]

File: docs/docker-compose.yml (hunk 17 above)
Comment (hieptranquoc) [reply docs/docker-compose.yml]: updated

**Review by hoatle — APPROVED** [review]
LGTM

**Pull request merged**